src/cargo/ops/cargo_doc.rs

```diff
@@ -53,7 +54,12 @@ pub fn doc(manifest_path: &Path,
         let target_dir = options.compile_opts.config.target_dir(&package);
         let path = target_dir.join("doc").join(&name).join("index.html");
         if fs::metadata(&path).is_ok() {
-            open_docs(&path);
+            match open_docs(&path) {
+                Ok(_) => (),
+                Err(_) => {
+                    println!("warning: could not determine a browser to open docs with.");
```

Comment:
maybe this could also print out the list of programs attempted? Ideally we wouldn't actually attempt programs that don't exist, but that can also perhaps be postponed.